modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -555,6 +781,30 @@ public void onFailure(Exception e) {
         });
     }
 
+    private void downsampleIndex(DownsampleAction.Request request, ActionListener<Void> listener) {
+        logger.trace(
+            "Data stream lifecycle issues request to downsample index [{}] to index [{}]",
+            request.getSourceIndex(),
+            request.getTargetIndex()
+        );
+        client.execute(DownsampleAction.INSTANCE, request, new ActionListener<>() {
+            @Override
+            public void onResponse(AcknowledgedResponse acknowledgedResponse) {
+                logger.info(
+                    "Data stream lifecycle successfully downsampled index [{}] to index [{}]",
+                    request.getSourceIndex(),
+                    request.getTargetIndex()
+                );
+                listener.onResponse(null);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
```

Comment:
Should this log about the failure also here?

Reply:
Absolutely. Note that I'm planning a follow-up PR where this logging logic is done in the error store itself.